astropy/modeling/physical_models.py

```diff
@@ -194,3 +196,125 @@ def lambda_max(self):
     def nu_max(self):
         """Peak frequency when the curve is expressed as power density."""
         return 2.8214391 * const.k_B * self.temperature / const.h
+
+
+class Drude1D(Fittable1DModel):
+    """
+    Drude model based one the behavior of electons in materials (esp. metals).
+
+    Parameters
+    ----------
+    amplitude : float
+        Peak value
+    x_0 : float
+        Position of the peak
+    fwhm : float
+        Full width at half maximum
+
+    Model formula:
+
+        .. math:: f(x) = A \\frac{(fwhm/x_0)^2}{((x/x_0 - x_0/x)^2 + (fwhm/x_0)^2}
+
+    Examples
+    --------
+
+    .. plot::
+        :include-source:
+
+        import numpy as np
+        import matplotlib.pyplot as plt
+
+        from astropy.modeling.models import Drude1D
+
+        fig, ax = plt.subplots()
+
+        # generate the curves and plot them
+        x = np.arange(7.5 , 12.5 , 0.1)
+
+        dmodel = Drude1D(amplitude=1.0, fwhm=1.0, x_0=10.0)
+        ax.plot(x, dmodel(x))
+
+        ax.set_xlabel('x')
+        ax.set_ylabel('F(x)')
+
+        ax.legend(loc='best')
+        plt.show()
+    """
+
+    amplitude = Parameter(default=1.0)
+    x_0 = Parameter(default=1.0)
+    fwhm = Parameter(default=1.0)
+
+    @staticmethod
+    def evaluate(x, amplitude, x_0, fwhm):
+        """
+        One dimensional Drude model function
+        """
+        return (
+            amplitude
+            * ((fwhm / x_0) ** 2)
+            / ((x / x_0 - x_0 / x) ** 2 + (fwhm / x_0) ** 2)
+        )
+
+    @staticmethod
+    def fit_deriv(x, amplitude, x_0, fwhm):
+        """
+        Drude1D model function derivatives.
+        """
+        d_amplitude = (fwhm / x_0) ** 2 / ((x / x_0 - x_0 / x) ** 2 + (fwhm / x_0) ** 2)
+        d_x_0 = (
```

Comment:
maybe it's just me, but this formatting is super difficult to read.

Reply:
I can change this, but suggestions would be needed.  I just use the black program to cleanup all the formatting so I don't have to worry about flake8 errors.

Reply:
yes, I recognize black from miles away now, not sure why they do this, it's hurting ones eyes and adds nothing for readability. I've added it to the upcoming meeting's agenda to come up with a solution that would work for contributors and codebase as well.

Reply:
That would be great.  There are just too many flake8 rules for me to by hand keep up.  I don't agree with all the choices black makes, but I sure love not having to worry about flake8 stuff anymore!

Reply:
autopep8 is another cmd line tool, it usually is more conservative than black. And almost all editors have plugins that notify you on the spot, but that indeed takes some time to get used to.